ipn/ipnlocal/local.go

```diff
@@ -2454,10 +2471,35 @@ func (b *LocalBackend) Start(opts ipn.Options) error {
 	if b.reconcilePrefsLocked(newPrefs) {
 		prefsChanged = true
 	}
+
+	// neither UpdatePrefs or reconciliation should change Persist
+	newPrefs.Persist = b.pm.CurrentPrefs().Persist().AsStruct()
+
+	// attempt to generate a new hardware attestion key if none exists
+	var ak key.HardwareAttestationKey
+	if newPrefs.Persist != nil {
+		ak = newPrefs.Persist.AttestationKey
+	}
+	if ak == nil || ak.IsZero() {
+		var err error
+		ak, err = key.NewHardwareAttestationKey()
+		if err != nil {
+			if !errors.Is(err, key.ErrUnsupported) {
+				b.logf("failed to create hardware attestation key: %v", err)
+			}
+		} else if ak != nil {
+			b.logf("using new hardware attestation key: %v", ak.Public())
+			if newPrefs.Persist == nil {
+				newPrefs.Persist = &persist.Persist{}
+			}
+			newPrefs.Persist.AttestationKey = ak
+			prefsChanged = true
+		}
+	} else {
+		b.logf("using existing hardware attestation key: %s", key.HardwareAttestationPublicFromPlatformKey(newPrefs.Persist.AttestationKey))
+	}
```

Comment:
this Start method is already super long. Can you pull this out into a method? Ideally something that can later be featuretagged away (see http://go/clientmod) for small builds that don't have hardware attestation hardware.